core/src/main/java/jenkins/model/NewViewLink.java

```diff
@@ -0,0 +1,52 @@
+package jenkins.model;
+
+import com.google.common.annotations.VisibleForTesting;
+import hudson.Extension;
+import hudson.model.Action;
+import hudson.model.TransientViewActionFactory;
+import hudson.model.View;
+import java.util.Collections;
+import java.util.List;
+
+@Extension
```

Comment:
Also recommend

```java
@Restricted(NoExternalUse.class)
```

since this is no API.

Reply:
+1